pkg/testcoverage/check.go

```diff
@@ -90,6 +96,21 @@ func Analyze(cfg Config, current, base []coverage.Stats) AnalyzeResult {
 	}
 }
 
+func detectOverrides(overrides []Override) (hasFileOverrides, hasPackageOverrides bool) {
+	for _, override := range overrides {
+		if strings.HasSuffix(override.Path, ".go") {
+			hasFileOverrides = true
+		}
+		if strings.HasPrefix(override.Path, "^") {
+			hasPackageOverrides = true
+		}
+		if hasFileOverrides && hasPackageOverrides {
```

Comment:
i don't have anything against of merging this `if`, but it seems unnecessary. these overrides are usually very small, lets say there is 100 of them (99.99% of users will have less then 10).  with this size, this `for` cycle will finish in `1ns` or less, so this `if` wont produce any noticeably benefits. and on another hand because this repo requires 100% coverage, this `if` will required to have additional tests case that needs to be added or ignored.